### DISCUSSION TITLE
How to track a specific TCP connection in Axum?

### DISCUSSION THREAD

**Discussion by alexpyattaev** [tokio-rs/axum#2952] [Q&A] [answered]
### Summary

I'm trying to track a particular TCP connection across calls to the endpoints in axum. I specifically want a per-connection request handling in my app (I am 100% certain that the compliant HTTP client will have keep-alive set). I know there is a mechanism to track remote port and IP that is discussed in https://github.com/tokio-rs/axum/issues/43, but I would much prefer having an actual connection tracking primitive (if there is one) rather than global state in the app tracking specific socket addresses and a mess of timeouts to clean that map. I imagine deep inside hyper actually tracks the specific TCP connections, so would appreciate if someone knows how to get to that info. Thanks!

### axum version

0.7.7

**Comment by jplatte** [comment, marked as answer]
See [`serve-with-hyper`](https://github.com/tokio-rs/axum/blob/main/examples/serve-with-hyper/src/main.rs) for an example of how to drive the TCP accept loop yourself, where you can then add whatever connection tracking code you want.

**Reply by alexpyattaev** [reply]
Oh that is exactly what I was looking for, thank you!
